WordPress/gutenberg
# Move search into inserter tabs

## What?
This moves the search box in the inserter inside the blocks and patterns tab and prioritizes the search by the tab (blocks search prioritizes blocks, for example).

## Why?
To keep the inserter open after users insert patterns as this will provide a better experience for assembling pages with patterns (see https://github.com/WordPress/gutenberg/issues/61051).

To achieve this we need to add a close button to the inserter.

To achieve this we need to move the search box below the tabs so that the close button can sit alongside the tabs like we do in other side panels, hence this PR.

## How?

- Move the search field below the tabs (see why above)
- Remove the auto focus on the search control ([reasoning](https://github.com/WordPress/gutenberg/pull/61108#issuecomment-2088347999))
- Always show the tabs even if there are no available contents in them
- Show a message when there are no blocks or patterns available

## Testing Instructions
1. Open the inserter
2. Check that the active tab is focused 
3. Search for something in the blocks tab
4. Should prioritize block results
5. Switch to patterns tab
6. Patterns from search should be shown

Do the same thing in the widget editor, site editor and post editor.

## Screenshots or screencast

https://github.com/WordPress/gutenberg/assets/107534/9a1d7585-d11a-4850-9085-426286c99c10
